uamqp/message.py

```diff
@@ -250,7 +250,10 @@ def data(self):
 
     @property
     def sequence(self):
-        if not self._message or not self._body:
+        try:
+            if not self._message or not self._body:
```

Comment:
BatchMessage doesn't have a _message attribute, so will go into except and return None.